compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -5838,10 +5740,72 @@ public void visit(BLangGroupExpr groupExpr) {
+    }
+
+    private List<String> getNamesOfRecordFields(List<RecordLiteralNode.RecordField> fields) {
+        List<String> fieldNames = new ArrayList<>();
+        for (RecordLiteralNode.RecordField field : fields) {
+            if (field.getKind() == NodeKind.RECORD_LITERAL_KEY_VALUE) {
+                BLangExpression key = ((BLangRecordLiteral.BLangRecordKeyValueField) field).key.expr;
+                if (key.getKind() == NodeKind.LITERAL) {
+                    fieldNames.add(((BLangLiteral) key).originalValue);
+                } else if (key.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
+                    fieldNames.add(((BLangSimpleVarRef) key).variableName.value);
+                }
+            }
+        }
+        return fieldNames;
+    }
+
+    private void updateFieldsOfRecordLiteral(List<RecordLiteralNode.RecordField> fields, List<String> fieldNames,
+                                             Map<String, BInvokableSymbol> defaultValues, Location pos) {
+        for (Map.Entry<String, BInvokableSymbol> entry : defaultValues.entrySet()) {
+            String fieldName = entry.getKey();
+            if (fieldNames.contains(fieldName)) {
+                continue;
+            }
+            fieldNames.add(fieldName);
+            BInvokableSymbol invokableSymbol = entry.getValue();
+            BLangInvocation closureInvocation = getInvocation(invokableSymbol);
+            BLangRecordLiteral.BLangRecordKeyValueField member = new BLangRecordLiteral.BLangRecordKeyValueField();
+            member.key = new BLangRecordLiteral.BLangRecordKey(ASTBuilderUtil.createLiteral(pos, symTable.stringType,
+                                                                                                            fieldName));
+            member.valueExpr = addConversionExprIfRequired(closureInvocation, closureInvocation.getBType());
+
+            fields.add(member);
+        }
+    }
+
+    private void updateFieldsOfRecordLiteral(BLangRecordLiteral recordLiteral,
+                                             List<RecordLiteralNode.RecordField> fields) {
+        BType type = Types.getReferredType(recordLiteral.getBType());
+        if (type.getKind() != TypeKind.RECORD) {
+            return;
+        }
+        List<String> fieldNames = getNamesOfRecordFields(fields);
+        Location pos = recordLiteral.pos;
+        updateFieldsOfRecordLiteral((BRecordType) type, fields, fieldNames, pos);
+    }
+
+    private void updateFieldsOfRecordLiteral(BRecordType recordType, List<RecordLiteralNode.RecordField> fields,
+                                             List<String> fieldNames, Location pos) {
+        Map<String, BInvokableSymbol> defaultValues = ((BRecordTypeSymbol) recordType.tsymbol).defaultValues;
+        updateFieldsOfRecordLiteral(fields, fieldNames, defaultValues, pos);
+
+        List<BType> typeInclusions = recordType.typeInclusions;
+
+        for (BType typeInclusion : typeInclusions) {
```

Comment:
The current tests don't test that the default is used from the including record rather that the included record, do they? See my comment on tests.

If someone changes the order by mistake it may not be captured via tests.

Reply:
Link for tests?

Reply:
Added a test with https://github.com/ballerina-platform/ballerina-lang/pull/37184/commits/fb9f549c2b690169aca53dd30eef490c30658c6e

What do you mean by `someone changes the order`?

Reply:
Can't quite recall now, but I think it was to do with if someone populated values from the included records first and then checked and used values from the including record.